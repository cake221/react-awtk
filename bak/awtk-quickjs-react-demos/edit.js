import React, { Component } from 'react';
import { Window, Button, AwtkRender, setParentWidget } from "../src"

class App extends Component {
  constructor(props){
    super(props);
  }

  componentDidMount () {
    console.log('APP DID MOUNT!');
  }

  render() {
    return (
      <Window
        ref = {
          (ref) => setParentWidget(ref, "win1")
        }
      >
        <Edit
          parent = { "win1" }
          text = { "" }
          name = { 'text[3-8]' }
          inputTips = { 'text[3-8]' }
          inputType = { TInputType.TEXT }
          style = {{
            selfLayout:{
              x:"10",
              y:"10",
              w:"228",
              h:"30"
            },
          }}
          onValueChanged = {
            (evt)=>{
              // console.log(edit.name + ' changed');
            }
          }
          onValueChanging = {
            (evt)=>{
              // console.log(edit.name + ' changed');
            }
          }
          textLimit = {[3, 8]}
        />
        <Edit
          parent = { "win1" }
          text = { "" }
          name = { 'int auto fix[1-100]' }
          inputTips = { 'int auto fix[1-100]' }
          inputType = { TInputType.INT }
          style = {{
            selfLayout:{
              x:"10",
              y:"50",
              w:"228",
              h:"30"
            },
          }}
          onValueChanged = {
            (evt)=>{
              // console.log(edit.name + ' changed');
            }
          }
          onValueChanging = {
            (evt)=>{
              // console.log(edit.name + ' changed');
            }
          }
          intLimit = {[1, 100, 1]}
          autoFix = { true }
        />
        <Edit
          parent = { "win1" }
          text = { '1.23' }
          name = { "float[1-10]" }
          inputTips = { "float[1-10]" }
          inputType = { TInputType.FLOAT }
          style = {{
            selfLayout:{
              x:"10",
              y:"90",
              w:"228",
              h:"30"
            },
          }}
          onValueChanged = {
            (evt)=>{
              // console.log(edit.name + ' changed');
            }
          }
          onValueChanging = {
            (evt)=>{
              // console.log(edit.name + ' changed');
            }
          }
          floatLimit = {[1, 10, 1]}
        />
        <Edit
          parent = { "win1" }
          text = { "" }
          name = { 'password' }
          inputTips = { 'password' }
          inputType = { TInputType.PASSWORD }
          style = {{
            selfLayout:{
              x:10,
              y:10+128,
              w:228,
              h:30
            },
          }}
          onValueChanged = {
            (evt)=>{
              // console.log(edit.name + ' changed');
            }
          }
          onValueChanging = {
            (evt)=>{
              // console.log(edit.name + ' changed');
            }
          }
          TextLimit = {}
        />
        <Edit
          parent = { "win1" }
          text = { "readonly" }
          name = { 'text' }
          inputTips = { 'text' }
          inputType = { TInputType.TEXT }
          style = {{
            selfLayout:{
              x:10,
              y:50+128,
              w:228,
              h:30
            },
          }}
          onValueChanged = {
            (evt)=>{
              // console.log(edit.name + ' changed');
            }
          }
          onValueChanging = {
            (evt)=>{
              // console.log(edit.name + ' changed');
            }
          }
          readonly = {true}
        />
        <Edit
          parent = { "win1" }
          text = { "" }
          name = { 'hex' }
          inputTips = { 'hex' }
          inputType = { TInputType.HEX }
          style = {{
            selfLayout:{
              x:10,
              y:90+128,
              w:228,
              h:30
            },
          }}
          onValueChanged = {
            (evt)=>{
              // console.log(edit.name + ' changed');
            }
          }
          onValueChanging = {
            (evt)=>{
              // console.log(edit.name + ' changed');
            }
          }
        />
        <Edit
          parent = { "win1" }
          text = { "" }
          name = { 'email' }
          inputTips = { 'email' }
          inputType = { TInputType.EMAIL }
          style = {{
            selfLayout:{
              x:10,
              y:130+128,
              w:228,
              h:30
            },
          }}
          onValueChanged = {
            (evt)=>{
              // console.log(edit.name + ' changed');
            }
          }
          onValueChanging = {
            (evt)=>{
              // console.log(edit.name + ' changed');
            }
          }
        />
      </Window>
    )
  }
}


AwtkRender(<App />, () => {
  console.log(' 渲染成功 AWTK ');
});


