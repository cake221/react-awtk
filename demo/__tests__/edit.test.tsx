import "../edit"

describe("test edit", ()=>{

  it("toMatchSnapshot", ()=>{
    expect(__AwtkSnapshot).toMatchSnapshot()
  })
})
