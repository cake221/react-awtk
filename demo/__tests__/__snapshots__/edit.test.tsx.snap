// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test edit toMatchSnapshot 1`] = `
Array [
  "getRootHostContext",
  "getChildHostContext",
  "shouldSetTextContent",
  "getChildHostContext",
  "shouldSetTextContent",
  "createInstance",
  "funtion  edit_create( parent, x, y, w, h ) :null, 0, 0, 0, 0",
  "funtion  widget_set_self_layout_params( widget, x, y, w, h ) :undefined, 10, 10, 228, 30",
  "funtion  widget_set_text_utf8( widget, text ) :undefined, ",
  "funtion  widget_set_name( widget, name ) :undefined, text[3-8]",
  "funtion  edit_set_input_type( widget, type ) :undefined, INPUT_TEXT",
  "funtion  edit_set_input_tips( widget, tips ) :undefined, text[3-8]",
  "funtion  widget_on( widget, type, on_event, ctx ) :undefined, EVT_VALUE_CHANGED, function (evt) {
        var target = src_1.TEvent.cast(evt).target;
        var instance = src_1.TEdit.cast(target);
        console.log(instance.name + ' changed');
        return TRet.OK;
    }, null",
  "funtion  widget_on( widget, type, on_event, ctx ) :undefined, EVT_VALUE_CHANGING, function (evt) {
        var target = src_1.TEvent.cast(evt).target;
        var instance = src_1.TEdit.cast(target);
        console.log(instance.name + ' changing');
        return TRet.OK;
    }, null",
  "funtion  edit_set_text_limit( widget, min, max ) :undefined, 3, 8",
  "finalizeInitialChildren",
  "getChildHostContext",
  "shouldSetTextContent",
  "createInstance",
  "funtion  edit_create( parent, x, y, w, h ) :null, 0, 0, 0, 0",
  "funtion  widget_set_self_layout_params( widget, x, y, w, h ) :undefined, 10, 50, 228, 30",
  "funtion  widget_set_text_utf8( widget, text ) :undefined, ",
  "funtion  widget_set_name( widget, name ) :undefined, int auto fix[1-100]",
  "funtion  edit_set_auto_fix( widget, auto_fix ) :undefined, true",
  "funtion  edit_set_input_type( widget, type ) :undefined, INPUT_INT",
  "funtion  edit_set_input_tips( widget, tips ) :undefined, int auto fix[1-100]",
  "funtion  widget_on( widget, type, on_event, ctx ) :undefined, EVT_VALUE_CHANGED, function (evt) {
        var target = src_1.TEvent.cast(evt).target;
        var instance = src_1.TEdit.cast(target);
        console.log(instance.name + ' changed');
        return TRet.OK;
    }, null",
  "funtion  widget_on( widget, type, on_event, ctx ) :undefined, EVT_VALUE_CHANGING, function (evt) {
        var target = src_1.TEvent.cast(evt).target;
        var instance = src_1.TEdit.cast(target);
        console.log(instance.name + ' changing');
        return TRet.OK;
    }, null",
  "funtion  edit_set_int_limit( widget, min, max, step ) :undefined, 1, 100, 1",
  "finalizeInitialChildren",
  "getChildHostContext",
  "shouldSetTextContent",
  "createInstance",
  "funtion  edit_create( parent, x, y, w, h ) :null, 0, 0, 0, 0",
  "funtion  widget_set_self_layout_params( widget, x, y, w, h ) :undefined, 10, 90, 228, 30",
  "funtion  widget_set_text_utf8( widget, text ) :undefined, 1.23",
  "funtion  widget_set_name( widget, name ) :undefined, float[1-10]",
  "funtion  edit_set_input_type( widget, type ) :undefined, INPUT_FLOAT",
  "funtion  edit_set_input_tips( widget, tips ) :undefined, float[1-10]",
  "funtion  widget_on( widget, type, on_event, ctx ) :undefined, EVT_VALUE_CHANGED, function (evt) {
        var target = src_1.TEvent.cast(evt).target;
        var instance = src_1.TEdit.cast(target);
        console.log(instance.name + ' changed');
        return TRet.OK;
    }, null",
  "funtion  widget_on( widget, type, on_event, ctx ) :undefined, EVT_VALUE_CHANGING, function (evt) {
        var target = src_1.TEvent.cast(evt).target;
        var instance = src_1.TEdit.cast(target);
        console.log(instance.name + ' changing');
        return TRet.OK;
    }, null",
  "funtion  edit_set_float_limit( widget, min, max, step ) :undefined, 1, 10, 1",
  "finalizeInitialChildren",
  "getChildHostContext",
  "shouldSetTextContent",
  "createInstance",
  "funtion  edit_create( parent, x, y, w, h ) :null, 0, 0, 0, 0",
  "funtion  widget_set_self_layout_params( widget, x, y, w, h ) :undefined, 10, 138, 228, 30",
  "funtion  widget_set_text_utf8( widget, text ) :undefined, ",
  "funtion  widget_set_name( widget, name ) :undefined, password",
  "funtion  edit_set_input_type( widget, type ) :undefined, INPUT_PASSWORD",
  "funtion  edit_set_input_tips( widget, tips ) :undefined, password",
  "funtion  widget_on( widget, type, on_event, ctx ) :undefined, EVT_VALUE_CHANGED, function (evt) {
        var target = src_1.TEvent.cast(evt).target;
        var instance = src_1.TEdit.cast(target);
        console.log(instance.name + ' changed');
        return TRet.OK;
    }, null",
  "funtion  widget_on( widget, type, on_event, ctx ) :undefined, EVT_VALUE_CHANGING, function (evt) {
        var target = src_1.TEvent.cast(evt).target;
        var instance = src_1.TEdit.cast(target);
        console.log(instance.name + ' changing');
        return TRet.OK;
    }, null",
  "finalizeInitialChildren",
  "getChildHostContext",
  "shouldSetTextContent",
  "createInstance",
  "funtion  edit_create( parent, x, y, w, h ) :null, 0, 0, 0, 0",
  "funtion  widget_set_self_layout_params( widget, x, y, w, h ) :undefined, 10, 178, 228, 30",
  "funtion  widget_set_text_utf8( widget, text ) :undefined, readonly",
  "funtion  widget_set_name( widget, name ) :undefined, text",
  "funtion  edit_set_readonly( widget, readonly ) :undefined, true",
  "funtion  edit_set_input_type( widget, type ) :undefined, INPUT_TEXT",
  "funtion  edit_set_input_tips( widget, tips ) :undefined, text",
  "funtion  widget_on( widget, type, on_event, ctx ) :undefined, EVT_VALUE_CHANGED, function (evt) {
        var target = src_1.TEvent.cast(evt).target;
        var instance = src_1.TEdit.cast(target);
        console.log(instance.name + ' changed');
        return TRet.OK;
    }, null",
  "funtion  widget_on( widget, type, on_event, ctx ) :undefined, EVT_VALUE_CHANGING, function (evt) {
        var target = src_1.TEvent.cast(evt).target;
        var instance = src_1.TEdit.cast(target);
        console.log(instance.name + ' changing');
        return TRet.OK;
    }, null",
  "finalizeInitialChildren",
  "getChildHostContext",
  "shouldSetTextContent",
  "createInstance",
  "funtion  edit_create( parent, x, y, w, h ) :null, 0, 0, 0, 0",
  "funtion  widget_set_self_layout_params( widget, x, y, w, h ) :undefined, 10, 218, 228, 30",
  "funtion  widget_set_text_utf8( widget, text ) :undefined, ",
  "funtion  widget_set_name( widget, name ) :undefined, hex",
  "funtion  edit_set_input_type( widget, type ) :undefined, INPUT_HEX",
  "funtion  edit_set_input_tips( widget, tips ) :undefined, hex",
  "funtion  widget_on( widget, type, on_event, ctx ) :undefined, EVT_VALUE_CHANGED, function (evt) {
        var target = src_1.TEvent.cast(evt).target;
        var instance = src_1.TEdit.cast(target);
        console.log(instance.name + ' changed');
        return TRet.OK;
    }, null",
  "funtion  widget_on( widget, type, on_event, ctx ) :undefined, EVT_VALUE_CHANGING, function (evt) {
        var target = src_1.TEvent.cast(evt).target;
        var instance = src_1.TEdit.cast(target);
        console.log(instance.name + ' changing');
        return TRet.OK;
    }, null",
  "finalizeInitialChildren",
  "getChildHostContext",
  "shouldSetTextContent",
  "createInstance",
  "funtion  edit_create( parent, x, y, w, h ) :null, 0, 0, 0, 0",
  "funtion  widget_set_self_layout_params( widget, x, y, w, h ) :undefined, 10, 258, 228, 30",
  "funtion  widget_set_text_utf8( widget, text ) :undefined, ",
  "funtion  widget_set_name( widget, name ) :undefined, email",
  "funtion  edit_set_input_type( widget, type ) :undefined, INPUT_EMAIL",
  "funtion  edit_set_input_tips( widget, tips ) :undefined, email",
  "funtion  widget_on( widget, type, on_event, ctx ) :undefined, EVT_VALUE_CHANGED, function (evt) {
        var target = src_1.TEvent.cast(evt).target;
        var instance = src_1.TEdit.cast(target);
        console.log(instance.name + ' changed');
        return TRet.OK;
    }, null",
  "funtion  widget_on( widget, type, on_event, ctx ) :undefined, EVT_VALUE_CHANGING, function (evt) {
        var target = src_1.TEvent.cast(evt).target;
        var instance = src_1.TEdit.cast(target);
        console.log(instance.name + ' changing');
        return TRet.OK;
    }, null",
  "finalizeInitialChildren",
  "createInstance",
  "funtion  window_create( parent, x, y, w, h ) :null, 0, 0, 0, 0",
  "appendInitialChild",
  "appendInitialChild",
  "appendInitialChild",
  "appendInitialChild",
  "appendInitialChild",
  "appendInitialChild",
  "appendInitialChild",
  "finalizeInitialChildren",
  "prepareForCommit",
  "appendNodeChild",
  "resetAfterCommit",
  "getPublicInstance",
  "funtion  widget_add_child( widget, child ) :undefined, [object Object]",
  "funtion  widget_add_child( widget, child ) :undefined, [object Object]",
  "funtion  widget_add_child( widget, child ) :undefined, [object Object]",
  "funtion  widget_add_child( widget, child ) :undefined, [object Object]",
  "funtion  widget_add_child( widget, child ) :undefined, [object Object]",
  "funtion  widget_add_child( widget, child ) :undefined, [object Object]",
  "funtion  widget_add_child( widget, child ) :undefined, [object Object]",
  "funtion  widget_layout( widget ) :undefined",
]
`;
